### PR TITLE
Handle login + auto re-login when bot account gets logged out

### PR DESCRIPTION
**Pull request by Jeto143**
Twitter API keys are supposed to be valid for a very long time, but for some reason they get invalidated very often (and a bit randomly). This PR addresses this issue by manually handling login and by detecting when the API key was invalidated in order to log back in again.

### CODE REVIEW

File: 5m5v-bot.js [hunk 1]
```diff
@@ -17,30 +17,57 @@ const languages = config.users.map(user => user.language);
...
+              const rettiwt = new Rettiwt({ apiKey: await getApiKey(user) });
+
+              await rettiwt.tweet.retweet(tweet.id);
+            }
+
+            console.log(`Retweeted tweet ${tweet.id} in ${language}:\n${tweet.fullText}`);
+          } catch (error) {
+            console.error(`Unable to retweet ${tweet.id} in ${language}: ${error.message}`);
+
+            if (error.constructor.name === 'RettiwtError' && error.code === 32) {
```
Comment (Jeto143) [inline 5m5v-bot.js:60]: This condition looks a bit ugly, but I couldn't find another way to discriminate it. (I'm using it in a personal project and it works.)

File: 5m5v-bot.js [hunk 2]
```diff
@@ -17,30 +17,57 @@ const languages = config.users.map(user => user.language);
...
-
 const tweetFilter = new TweetFilter(config.exclude, languages);
-const rettiwt = new Rettiwt({ apiKey: config.users[0].api_key });
 
-console.log(isDryRun ? 'Looking for new tweets (dry run)...' : 'Looking for new tweets...');
+async function getApiKey(user) {
+  if (!user.apiKey) {
+    console.log(`Logging in as ${user.username}...`);
+
+    user.apiKey = await new Rettiwt().auth.login(user.email, user.username, user.password);
```
Comment (Headscracher) [inline 5m5v-bot.js:34]: Even tho the project is meant for internal usage as we're having downloads of the npm package we should probably if the first user, used for fetching is set and if not send an error message

**Review by Headscracher — APPROVED** [review]

File: 5m5v-bot.js (hunk 2 above)
Comment (Jeto143) [reply 5m5v-bot.js:34]: Good point! Fixed [here](https://github.com/veganhacktivists/5m5v-bot/pull/5/files/12e533444ad95a1ea6e46ce09819d2db27b55e84..741b424b577d668bdb87c69f81396bbe8418ebf5).

File: 5m5v-bot.js [hunk 3]
```diff
@@ -13,6 +13,14 @@ try {
   throw `Unable to load config file: ${error.message}`;
 }
 
+if (!('users' in (config ?? [])) || config.users.length === 0) {
+  throw 'No users defined in config file';
+}
+
+if (!config.users.every(user => ['language', 'email', 'username', 'password'].every(key => key in (user ?? [])))) {
+  throw 'At least one user is missing required fields in config file';
```
Comment (Headscracher) [inline 5m5v-bot.js:21]: I think in general it's not a bad thing to check for those fields for every user entry, but the error message should be altered. Something like "You are missing fields for one of your user entries"

**Review by Headscracher — APPROVED** [review]

**Pull request merged**